scripts/SMD_chip_package_rlc-etc/size_definitions/size_capacitor_chip_smaller_0603.yaml

```diff
@@ -0,0 +1,12 @@
+C_0402:
+    code_imperial: "0402"
+    code_metric: "1005"
+    body_length:
+      minimum: 0.9
+      maximum: 1.1
+    body_width:
+      minimum: 0.4
+      maximum: 0.6
+    terminator_spacing_min: 0.3
+    terminator_spacing_max: 0.65
```

Comment:
That is wrong. And uses the old style.
`terminator_spacing: 0.1 .. 0.3` according to the spec.

Reply:
Note we're looking at terminal length in some definitions and terminal spacing in others. It's been a while so I didn't catch that at first. With L=0.9 to 1.1mm and T=0.1 to 0.3mm, the spacing I calculate is 0.3mm min and 0.9mm max. As mentioned above, if there is a notable change to what I expect is a very common footprint we should carefully evaluate and be prepared to answer any questions from users of the library. Or leave ourselves a comment for later and wait for v6's library to do it. I'm just saying to be thoughtful if changing high-impact footprints.

Note that this dimension style was carried over from https://github.com/pointhi/kicad-footprint-generator/blob/master/scripts/SMD_chip_package_rlc-etc/size_definitions/size_default_chip_devices_smaller_0603.yaml.

Perhaps we should look at this a bit closer. Probably we should consolidate on the same style everywhere where possible, and I see a lot of hits for 'spacing' if I search for that term in the YAML files. Many of them are in comments, however. I'm thinking to remove all the comment lines and then see if we can consolidate on terminal length and not spacing. That will require looking into the code a bit so that's why I'm asking first.

Reply:
OK. A little bit more about this. The 0.1-0.3mm you wrote above is terminal length. The given dimensions of 0.3-0.65mm are terminal spacing. Both are in IPC-SM-782 so one would think we are good. I believe what you're pointing out is a difference between terminal _spacing_ vs terminal _length.

That being said, we do end up with slightly different footprints if we use spacing vs length. Here is current:
- Pad centers: +/-0.45mm
- Pad width: 0.62mm
- Pad inside: 0.28mm
- Pad outside: 1.52mm

And if I use terminal length from IPC instead of spacing:
- Pad centers: +/-0.48mm
- Pad width: 0.56mm
- Pad inside: 0.4mm
- Pad outside: 1.52mm

Here the pad spacing is farther apart. Above it was closer. I'm wondering if we should standardize on spacing or length for the IPC body sizes, but I don't know which is better. Ugh. Again, I'll ask your thoughts.

Reply:
> I believe what you're pointing out is a difference between terminal spacing vs terminal _length.
Yes, that is what my original comment was about. It is inconsistent with other definitions using `terminal_length`.

> I'm just saying to be thoughtful if changing high-impact footprints.

:+1: 

> Perhaps we should look at this a bit closer. Probably we should consolidate on the same style everywhere where possible, and I see a lot of hits for 'spacing' if I search for that term in the YAML files. Many of them are in comments, however. I'm thinking to remove all the comment lines and then see if we can consolidate on terminal length and not spacing. That will require looking into the code a bit so that's why I'm asking first.

I would love to see the same yaml style used everywhere. That makes reviews like this much easier. Less confusion. Which we should prevent with high-impact-footprints.

Looking at the Kemet Datasheet again, they get for IPC-Nominal
*   Pad centers: +/-0.45mm
*   Pad width: 0.62mm
*   Pad inside: 0.28mm
*   Pad outside: 1.52mm

I also looked for Wuerth, they have good datasheets and Sales-reps I could call: https://www.we-online.de/katalog/datasheet/885012205012.pdf
*   Pad centers: +/-0.75mm
*   Pad width: 0.5mm
*   Pad inside: 0.5mm
*   Pad outside: 1.50mm

One more Data point from taiyo-yuden. They clearly sell parts with different `terminal_length` values.
![image](https://user-images.githubusercontent.com/17516572/93258065-0d677980-f79e-11ea-9205-07658c94cb55.png)

Reply:
Heh. So the Kemet is similar to using terminal spacing and the Wuerth is similar to using terminal length.

In the Taiyo Yuden datasheet, it looks like that is different heights, not terminal widths. Part height is T.

Here are some posts from Tom Hausherr, who really, knows, in order of date. I haven't fully digested it yet but it looks like there was some grappling with this also and they settled on tweaking the IPC 7351 numbers to get a single footprint:
https://www.pcblibraries.com/forum/standard-chip-component-dimensions_topic1729.html
https://www.pcblibraries.com/forum/new-2016-solder-joint-goals_topic1921.html
https://www.pcblibraries.com/forum/ipcjstd001-chip-component-solder-joints_topic2588.html

Reply:
I was to quick to judge the Taiyo Yuden picture - my bad. But the height also seems to be an important thing.

From https://www.pcblibraries.com/forum/standard-chip-component-dimensions_topic1729.html
> Good luck with finding a single footprint to successfully accommodate every 0603 package. The Height is the main driver of this component lead form.
 > Issue: if you use a 0603 land pattern considering the highest package, you may be dumping too much solder on an 0603 that is 50% of that height, introducing potential tomb-stoning. 

From my personal experience:
This is especially a problem with capacitors. Chip-resistors seem to have mostly the same height.

This is an optimization problem:
* From all commercially available 0402 caps most should show perfect solder results
* The distance between pads should be maxed
* The courtyard should be minified
* Users should not question the footprint on first sight

Reply:
Yes, but the 5th comment from the 2nd link, noting that I posted them in order or date above, says that height was ruled out because it was too specific and wouldn't allow multiple sourcing with only one ideal footprint. Again, that was from 2016 and it sounds like that is the preferred direction but for now more sloppiness is allowed. The last link I posted is less than a year old and seems to confirm that.

I agree with what you said. The generators are available if anyone wants to utilize them, but let's try to go for the 80/20 rule and support at least 80% of users and situations without question.

I would like to change from using terminal spacing to terminal length for the 0402 part, in combination with the IPC size info. See my edited initial reply above where I added the existing footprint info. Are you OK with that?

Reply:
Agree. IPC size info and use `terminal_length` everywhere.

Reply:
Done.